net/respondd/README.md

```diff
@@ -0,0 +1,35 @@
+respondd providers are C modules (shared objects). These modules should include
+<respondd.h>, which provides the following definitions:
+
+        typedef struct json_object * (*respondd_provider)(void);
+
+        struct respondd_provider_info {
+                const char *request;
+                const respondd_provider provider;
+        };
+
+        extern const struct respondd_provider_info respondd_providers[];
+
+
+The module must define the array `respondd_providers`, e.g. like this:
+
+        static struct json_object * respondd_provider_nodeinfo(void) {
+                ...
+        }
+
+        static struct json_object * respondd_provider_statistics(void) {
+                ...
+        }
+
+        const struct respondd_provider_info respondd_providers[] = {
+                {"nodeinfo", respondd_provider_nodeinfo},
+                {"statistics", respondd_provider_statistics},
+                {}
+        };
+
+The array must be terminated with an empty entry. The provider for each
+request type must return a JSON object. This JSON object must have exactly 1
```

Comment:
I think we definitely need a reference to JSON-C here, as we can not assume everybody is familiar with it:
```
…must return a [JSON-C] JSON object…

[JSON-C]: https://github.com/json-c/json-c/wiki
```